paddle/fluid/operators/reader/create_double_buffer_reader_op.cc

```diff
@@ -171,16 +168,26 @@ void DoubleBufferReader::ReInit() {
   StartPrefetcher();
 }
 
+bool DoubleBufferReader::HasNext() const {
+  while (!channel_->IsClosed() && !channel_->CanReceive()) {
```

Comment:
...We should use a conditional variable to wait here.

Reply:
I will fix it in further PRs.